src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,115 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.NoSuchElementException;
+
 public class WorkWithFile {
+    private static final String SPLITERATOR = ",";
+    private static final String CASE_BUY = "buy";
+    private static final String CASE_SUPPLY = "supply";
+    private static final String RESULT = "result";
+    private static final String EXCEPTION_MESSAGE = "Key doesn't found";
+    private static final String PARSE_EXCEPTION_MESSAGE = "Something went wrong";
+    private static final int SUPPLY_VALUE = 1;
+    private static final int BUY_VALUE = 3;
+    private static final int KEY = 0;
+    private static final int KEY_VALUE = 1;
+    private static final String SPLIT_BY_NEW_LINE = System.lineSeparator();
+    private static final int END_FILE_CODE = -1;
+
     public void getStatistic(String fromFileName, String toFileName) {
```

Comment:
To improve code structure and readability, consider breaking down the process into three distinct methods: readFile(), generateReport(), and writeToFile(). Each method will have a clear responsibility, enhancing maintainability. For instance:

1. Use readFile(fromFileName) to fetch and return the file's contents.
2. Pass the content to generateReport(data) to create the desired report.
3. Finally, call writeToFile(toFileName, report) to save the report.

Applying this structure to your current code would not only simplify the getStatistic method but also follow good coding practices. Here's a quick outline:

    String data = readFile(fromFileName);
    String report = generateReport(data);
    writeToFile(toFileName, report);

This approach will keep your code organized and make each part easier to test and debug.

Reply:
Then you should agree between the reviewers on a common decision because one says one thing, and you say another, what should you do then?

Reply:
You are right that reviewers should reach a common decision when disagreeing. Please ping us in the chat to sort it out in cases like this 🤝(we don't receive notifications about replies in reviewed PRs)